app/scripts/lib/metaRPCClientFactory.test.js

```diff
@@ -131,4 +131,16 @@ describe('metaRPCClientFactory', () => {
       },
     });
   });
+
+  it('should be able to handle no message within TIMEOUT secs', async () => {
+    const streamTest = createThoughStream();
+    const metaRPCClient = metaRPCClientFactory(streamTest);
+    metaRPCClient.foo('bad', (error, _) => {
+      expect(error.message).toStrictEqual('No response from RPC');
+    });
+
+    const delay = (ms) => new Promise((res) => setTimeout(res, ms));
+
+    await delay(11000);
```

Comment:
Instead of putting an `expect` in a callback (which may never get called, thus creating a false positive), what do you think about creating a promise that gets resolved when this callback is called, then awaiting that promise? Something like:

``` suggestion
    const promise = new Promise((resolve) => {
      metaRPCClient.foo('bad', (error) => {
        resolve(error);
      });
    });

    const delay = (ms) => new Promise((res) => setTimeout(res, ms));

    await delay(11000);

    const error = await promiseForError;
    expect(error.message).toStrictEqual('No response from RPC');
```

Reply:
@mcmire I see this pattern used in this test (expect in callback). I'm torn between fixing all the occurrences or creating a new PR to fix them...I'm leaning towards creating a new PR to fix them all. Thoughts?

Reply:
Actually I wonder if we could just change this one test. My main concern is that by adding a delay like this you're adding 11s to the time it takes to run the test suite. Considering that the test suite already takes quite a bit of time to run anything we can do to keep it from growing significantly would help in the long run I think.